core/src/main/java/com/sap/ai/sdk/core/AiCoreService.java

```diff
@@ -25,123 +24,94 @@
-    val dest = baseDestinationHandler.apply(this);
-    val builder = builderHandler.apply(this, dest);
-    if (!deploymentId.isEmpty()) {
-      destinationSetUrl(builder, dest);
-      destinationSetHeaders(builder);
-    }
-    return builder.build();
+  AiCoreService(@Nonnull final Supplier<HttpDestination> baseDestinationResolver) {
+    this.baseDestinationResolver = baseDestinationResolver;
+    this.deploymentResolver = new DeploymentResolver(this);
   }
 
   /**
-   * Update and set the URL for the destination.
+   * Set a specific base destination. This is useful when loading a destination from the BTP
+   * destination service or some other source.
    *
-   * @param builder The destination builder.
-   * @param dest The original destination reference.
-   */
-  protected void destinationSetUrl(
-      @Nonnull final DefaultHttpDestination.Builder builder, @Nonnull final Destination dest) {
-    String uri = dest.get(DestinationProperty.URI).get();
-    if (!uri.endsWith("/")) {
-      uri = uri + "/";
-    }
-    builder.uri(uri + "v2/inference/deployments/%s/".formatted(deploymentId));
-  }
-
-  /**
-   * Update and set the default request headers for the destination.
+   * <p><b>Note:</b> For typical scenarios, the destination is expected to have the {@code /v2/}
+   * base path set. But for special cases a different base path may be required (e.g. when consuming
+   * AI Core via some proxy that expects a different base path).
    *
-   * @param builder The destination builder.
+   * @param destination The base destination to be used for AI Core service calls.
+   * @return A new AI Core Service object using the provided destination as basis.
    */
-  protected void destinationSetHeaders(@Nonnull final DefaultHttpDestination.Builder builder) {
-    builder.property(AI_RESOURCE_GROUP, resourceGroup);
+  @Nonnull
+  public AiCoreService withBaseDestination(@Nonnull final HttpDestination destination) {
+    return new AiCoreService(() -> DestinationResolver.fromCustomBaseDestination(destination));
   }
 
   /**
-   * Set a specific base destination.
+   * Get the base destination for AI Core service calls. This destination won't have any resource
+   * group set.
    *
-   * @param destination The destination to be used for AI Core service calls.
-   * @return The AI Core Service based on the provided destination.
+   * @return The base destination.
+   * @throws DestinationAccessException If there was an issue creating the base destination, e.g. in
+   *     case of invalid credentials.
+   * @throws DestinationNotFoundException If there was an issue creating the base destination, e.g.
+   *     in case of missing credentials.
+   * @see #withBaseDestination(HttpDestination)
    */
   @Nonnull
-  public AiCoreService withDestination(@Nonnull final HttpDestination destination) {
-    baseDestinationHandler = service -> destination;
-    return this;
+  public HttpDestination getBaseDestination()
```

Comment:
(Question/Minor)

What's the use-case for this method being public for users?

Reply:
This is what we could recommend for anyone who want to execute requests against AI Core manually e.g. using our HTTP client. That way the base path + client type header are already set, and it's much easier, compared to using the `ServiceBindingDestinationLoader` logic directly.

Reply:
Okay, while reviewing I wasn't aware, you are promoting `HttpDestination` as potential adapter between the APIs